### PR TITLE
Minor changes (fix #21)

### LINKED ISSUE #21: Add option to use existing transport when 'create_transport' is used
Hello,

it would be cool to use an existing transport request when the `create_transport` is used. It should be determine to use the existing request when the name of the request match with the given one from the `transport_text` option. Otherwise it should create a new one with the given name.

Best regards

Max

### PR DESCRIPTION
**Pull request by mxschmitt**
This PR includes:
- added using the locked transport if the BSP Application is locked
- added reusing of the created transport request (fix #21)
- updated packages
- changed empty file for git to .gitkeep

If you have any questions or I documented to less feel free to ping me.

### CODE REVIEW

**Comment by mxschmitt** [comment]
@pfefferf why?
I've created two issues, both were really good documented.

This PR also adds support for a feature which is optional, you haven't said any feedback to it why or whats the problem is.
Do you don't have the time and only want to close them down?
I definitely don't want to create a fork for the time being, rather integrate it into your version with it.

**Comment by pfefferf** [comment]
Sorry for closing the PR by fault (damn fat fingers on mobile device). And yes, I closed the "issues", cause I do not have the time to implement some requirements at the moment. A PR is the best way for that, as you have done it.

I have added some remarks to your PR. 
Also I am not very happy with the "find an open request for the user" functionality, because I do not see the real use case for bigger projects, but as it is optional and you have the requirement, I will accept the PR after the remarks are fixed.

**Comment by mxschmitt** [comment]
Okay cool. :)
Instead of closing the issues, you could let them open and the labels `feature` and `help wanted` to it. So other people can implement it via a PR.

File: tasks/lib/filestore.js [hunk 1]
```diff
@@ -672,6 +672,17 @@ FileStore.prototype.syncFile = function (sFile, sModif, sCwd, fnCallback) {
 
     me._sendRequest(oRequest, function (oResponse) {
         if (oResponse.error) {
+            if (me._oOptions.ui5.transport_use_locked) {
```
Comment (pfefferf) [inline tasks/lib/filestore.js]: Again as said before. Doing that specific stuff in that central function is "bullshit".

**Comment by mxschmitt** [comment]
image
Better? In the `sendRequest ` is the only way to check that in my opinion, because we do recursion to retry the request.

**Comment by pfefferf** [comment]
Nope, because in detail nothing has changed. the sendRequest function still does the error response handling (now just encapsulated in a function). So the caller is not able to steer how it should be reacted on errors. Of course it can be done as in the proposed way, but in case of additional case the coding must be called out of the sendRequest function. Better: Error handling done by the caller directly or by injection an error handler function.

**Comment by mxschmitt** [comment]
Hi, I tried to investigate some more into this issue, but I didn't have an idea how to solve this.
I understand you, that this isn't very clean. But i the end, the send `_sendRequest` will be called only once and  injecting there an error handler is also not very trivial. Or do you have any other ideas?

**Comment by pfefferf** [comment]
I have merged your changes now. Will look at the coding later for possible refactoring options regarding the discussed point (maybe I promisify the thing).
Will provide a updated npm package version, so that you are able to use it in a "standard way".

**Comment by mxschmitt** [comment]
Cool, I thought also about changing the es5 syntax to es6, so we can use classes instead of prototype functions. Does this make sense in your opinion?

**Comment by pfefferf** [comment]
Maybe in a later step when the support of Node 4 ends (end of April this year). Node 4 has still some ES6 issues.

**Pull request merged**